From e3ab5276c51a2128d4816ac1ed75ca8262d99b3c Mon Sep 17 00:00:00 2001
From: "Michael W. Bringmann" <mwb@linux.ibm.com>
Date: Tue, 24 Mar 2020 15:36:01 -0700
Subject: [PATCH] lsdevinfo: Fix bash command substitution warning

References: bsc#1171892 ltc#185700
Upstream: pending - expected 1.3.8
Git-commit: e3ab5276c51a2128d4816ac1ed75ca8262d99b3c

This patch is to remove the warning,

/usr/sbin/lsdevinfo: line 278: warning: command substitution: ignored null
byte in input.

observed in some shells.  This warning appears to be a new feature
in Bash-4.4.

Signed-off-by: Michael W. Bringmann <mwb@linux.ibm.com>
Signed-off-by: Tyrel Datwyler <tyreld@linux.ibm.com>
---
 scripts/lsdevinfo | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/lsdevinfo b/scripts/lsdevinfo
index 85019e9d9923..1d9597bc4bc9 100755
--- a/scripts/lsdevinfo
+++ b/scripts/lsdevinfo
@@ -275,7 +275,7 @@ for pci_dev in $($LS -d /proc/device-tree/pci* 2> /dev/null); do
 	# get the physical location 
 	physloc=$(tr -d '\0' < $dev/ibm,loc-code)
 	type="$($OD -t x2 $dev/vendor-id $dev/device-id |
-		$CUT -f3,5 -d ' ' -s --output-delimiter='')"
+		$CUT -f3,5 -d ' ' -s --output-delimiter='' | tr -d '\0')"
 	uniquetype="adapter/pci/$type"
 	class="adapter"
 	subclass="pci"
-- 
2.26.2

